plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -76,17 +80,119 @@ function isJobQueryArg(arg: any): arg is JobQueryArgs {
   return arg.id !== undefined;
 }
 
+// Sort and filtering
+function sortJobs(
+  jobs: Job[],
+  sortBy: SortOption = "ID",
+  sortDirection: SortDirection = "ASC"
+): Job[] {
+  jobs.sort((a, b) => {
+    let result;
+
+    switch (sortBy) {
+      case "ID":
+        result = sortJobById(a, b);
+        break;
+      case "STATUS":
+        result = sortJobByStatus(a, b);
+        break;
+      case "LAST_RUN":
+        result = sortJobByLastRun(a, b);
+        break;
+      default:
+        result = 0;
+    }
+
+    const direction = sortDirection === "ASC" ? 1 : -1;
+    return result * direction;
+  });
+
+  return jobs;
+}
+
+function sortJobById(a: Job, b: Job): number {
+  return a.id.localeCompare(b.id);
+}
+
+function sortJobByStatus(a: Job, b: Job): number {
+  return (
+    JobStates[a.scheduleStatus].sortOrder -
+    JobStates[b.scheduleStatus].sortOrder
+  );
+}
+
+function sortJobByLastRun(a: Job, b: Job): number {
+  return (
+    JobStatus[a.lastRunStatus.status].sortOrder -
+    JobStatus[b.lastRunStatus.status].sortOrder
+  );
+}
+
+function filterJobsById(
+  jobs: MetronomeJobResponse[],
+  filter: string | null
+): MetronomeJobResponse[] {
+  if (filter === null) {
+    return jobs;
+  }
+
+  return jobs.filter(({ id }) => id.indexOf(filter) !== -1);
+}
+
+function filterJobsByNamespace(
+  jobs: MetronomeJobResponse[],
+  namespace: string | null
+): MetronomeJobResponse[] {
+  if (!namespace) {
+    return jobs;
+  }
+
+  return jobs.filter(job => {
+    return job.id.startsWith(namespace);
+  });
+}
+
 export const resolvers = ({
+  fetchJobs,
   fetchJobDetail,
   pollingInterval
 }: ResolverArgs): IResolvers => ({
   Query: {
     jobs(
-      _obj = {},
-      _args: GeneralArgs = {},
+      _parent = {},
+      args: GeneralArgs = {},
       _context = {}
     ): Observable<JobConnection | null> {
-      return Observable.of(null);
+      const {
+        sortBy = "ID",
+        sortDirection = "ASC",
+        filter = null,
+        namespace = null
+      } = args as JobsQueryArgs;
+
+      const pollingInterval$ = Observable.timer(0, pollingInterval);
+      const responses$ = pollingInterval$.exhaustMap(fetchJobs);
+
+      const jobsInNamespace$ = responses$.map(jobs =>
+        filterJobsByNamespace(jobs, namespace)
```

Comment:
we could have created a curried function here :trollface: 
```js
const jobsInNamespace$ = responses$.map(getNamespaceJobsFilter(namespace))
```

Reply:
I would argue that this is unnecessarily complicated to read, but if you like 🤷‍♂️

Reply:
no-no, that was more a comment to troll @Fabs, please ignore

Reply:
I actually started with the partially applied function, then I realized the elegance might have a negative impact on comprehension.

Reply:
![023](https://user-images.githubusercontent.com/530632/41584645-23e64898-73a8-11e8-9b12-5328505b40d8.jpg)